zingolib/src/test_framework/mocks.rs

```diff
@@ -1,67 +1,126 @@
 
-pub struct TransparentNoteBuilder {
-    address: Option<String>,
-    txid: Option<TxId>,
-    output_index: Option<u64>,
-    script: Option<Vec<u8>>,
-    value: Option<u64>,
-    spent: Option<Option<(TxId, u32)>>,
-    unconfirmed_spent: Option<Option<(TxId, u32)>>,
+    (
+        extsk,
+        PreparedIncomingViewingKey::new(&fvk.fvk().vk.ivk()),
+        addr,
+    )
 }
-#[allow(dead_code)] //TODO:  fix this gross hack that I tossed in to silence the language-analyzer false positive
-impl TransparentNoteBuilder {
-    pub fn new() -> Self {
-        Self::default()
-    }
-    // Methods to set each field
-    build_method!(address, String);
-    build_method!(txid, TxId);
-    build_method!(output_index, u64);
-    build_method!(script, Vec<u8>);
-    build_method!(value, u64);
-    build_method!(spent, Option<(TxId, u32)>);
-    build_method!(unconfirmed_spent, Option<(TxId, u32)>);
-
-    // Build method
-    pub fn build(self) -> TransparentNote {
-        TransparentNote::from_parts(
-            self.address.unwrap(),
-            self.txid.unwrap(),
-            self.output_index.unwrap(),
-            self.script.unwrap(),
-            self.value.unwrap(),
-            self.spent.unwrap(),
-            self.unconfirmed_spent.unwrap(),
-        )
-    }
+
+pub fn default_txid() -> zcash_primitives::transaction::TxId {
+    zcash_primitives::transaction::TxId::from_bytes([0u8; 32])
+}
+pub fn default_zaddr() -> (
+    ExtendedSpendingKey,
+    PreparedIncomingViewingKey,
+    PaymentAddress,
+) {
+    zaddr_from_seed([0u8; 32])
 }
 
-impl Default for TransparentNoteBuilder {
-    fn default() -> Self {
-        TransparentNoteBuilder {
-            address: Some("default_address".to_string()),
-            txid: Some(TxId::from_bytes([0u8; 32])),
-            output_index: Some(0),
-            script: Some(vec![]),
-            value: Some(0),
-            spent: Some(None),
-            unconfirmed_spent: Some(None),
+use rand::{rngs::OsRng, Rng};
+use sapling_crypto::{
+    note_encryption::PreparedIncomingViewingKey, zip32::ExtendedSpendingKey, PaymentAddress,
+};
+
+pub fn random_txid() -> zcash_primitives::transaction::TxId {
+    let mut rng = OsRng;
+    let mut seed = [0u8; 32];
+    rng.fill(&mut seed);
+    zcash_primitives::transaction::TxId::from_bytes(seed)
+}
+pub fn random_zaddr() -> (
+    ExtendedSpendingKey,
+    PreparedIncomingViewingKey,
+    PaymentAddress,
+) {
+    let mut rng = OsRng;
+    let mut seed = [0u8; 32];
+    rng.fill(&mut seed);
+
+    zaddr_from_seed(seed)
+}
+
+// Sapling Note Mocker
+mod sapling_note {
+
+    use sapling_crypto::value::NoteValue;
+    use sapling_crypto::Note;
+    use sapling_crypto::PaymentAddress;
+    use sapling_crypto::Rseed;
+
+    use super::default_zaddr;
+
+    /// A struct to build a mock sapling_crypto::Note from scratch.
+    /// Distinguish sapling_crypto::Note from crate::wallet::notes::SaplingNote. The latter wraps the former with some other attributes.
```

Comment:
if you add [`thingy-path`] then it links it in cargo doc whihc is a habit we should get used to
```suggestion
    /// Distinguish [`sapling_crypto::Note`] from [`zingolib::wallet::notes::SaplingNote`]. The latter wraps the former with some other attributes.
```